### PR TITLE
[Language server] vs code client

### PR DESCRIPTION
**Pull request by ysndr**
Add VS Code extension and document how to install

### CODE REVIEW

**Review by yannham — APPROVED** [review]
I left a few comments about the README. I'll patch it with additional stuff (how to install the LSP directly with Cargo e.g.) after this is merged.

File: lsp/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Nickel Language Server
+
+NLS is a language server for the nickel programming language.  Making use of the interpreter libraries directly, allows nls to offer error messages, type hints, and autocompletion right in your favorite lsp-enabled editor.
```
[the remark this answers is not in the document]
Comment (ysndr) [reply lsp/README.md:3]: Ah no not really, I agree

**Comment by yannham** [comment]
@ysndr is there anything left to do before merging this?

**Comment by ysndr** [comment]
@yannham No were good. Maybe @Avi-D-coder can contribute the instructions for running with tracing using neovim-lsp, but its not necessary for this PR.

**Pull request merged**